src/service/BookingService.java

```diff
@@ -53,16 +53,15 @@ public Integer saveBooking(BookingData bookingData )  {
 		if (entryDate.isBefore(LocalDateTime.now())) {
 			throw new IllegalArgumentException("Entry date can't be before today");
 		}
-		
-		Duration duration=Duration.between(entryDate, departureDate);
-		BigDecimal days=new BigDecimal(duration.toDays());
-		
-		BookingData bookingDataWithPrice=new BookingData(entryDate,departureDate,paymentMethod,bookingPrice(days));
+	
+		BookingData bookingDataWithPrice=new BookingData(entryDate,departureDate,bookingPrice(entryDate,departureDate),paymentMethod);
```

Comment:
There is no need to create a new BookingData since the method already passes one as a parameter, so please delete the object `bookingDataWithPrice`.